application/backend/src/main/resources/apps/typerefinery/components/widgets/search/clientlibs/functions.js

```diff
@@ -0,0 +1,78 @@
+window.Typerefinery = window.Typerefinery || {};
+window.Typerefinery.Components = Typerefinery.Components || {};
+window.Typerefinery.Components.Widgets = Typerefinery.Components.Widgets || {};
+window.Typerefinery.Components.Widgets.Table = Typerefinery.Components.Widgets.Table || {};
+window.Typerefinery.Components.Widgets.Search = Typerefinery.Components.Widgets.Search || {};
+window.Typerefinery.Components.Widgets.Treeview = Typerefinery.Components.Widgets.Treeview || {};
+window.Typerefinery.Components.Widgets.Treeview.Instances = Typerefinery.Components.Widgets.Treeview.Instances || {};
+
+; (function (ns, componentNs, tableNs, treeViewInstanceNs, document, window) {
```

Comment:
having this component just raise events will minimise integrating all the possible combinations here.

this should just raise a global page event with topic specified.